infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/CommonTypeDefinitions.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.restapi.types;
+
+import java.util.Optional;
+import java.util.function.Function;
+import tech.pegasys.teku.infrastructure.http.HttpErrorResponse;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class CommonTypeDefinitions {
+  public static final DeserializableTypeDefinition<String> STRING_TYPE =
+      new StringBasedPrimitiveTypeDefinition<>(
+          Function.identity(), Function.identity(), Optional.empty(), Optional.empty());
+
+  public static final DeserializableTypeDefinition<UInt64> UINT64_TYPE =
+      new StringBasedPrimitiveTypeDefinition<>(
+          UInt64::valueOf,
+          UInt64::toString,
+          Optional.of("unsigned 64 bit integer"),
+          Optional.empty());
```

Comment:
i think currently we set the format as "$uint64"

Reply:
We might be able to do these as builder, the other optional that we should be able to specify is the pattern - we use them a lot in bytes.
The other thing would be example - also optional but should be less optional and more often set (used in the generated examples on swagger)

Reply:
Turns out it's just `uint64` currently.

Reply:
Yeah quite possibly will wind up turning this into a builder as things get more complex.  I'm just hesitant to add too much overhead until I get to a point where I can see which types we really will need and what it looks like to define them.  Since I'm hoping that `CommonTypeDefinitions` is the only place we need to construct these we may not need to go too crazy tidying up the construction, but not really sure yet.